arcade/sprite/base.py

```diff
@@ -16,6 +16,16 @@
 class BasicSprite:
     """
     The absolute minimum needed for a sprite.
+
+    .. warning:: Beginners should see :py:class:`~arcade.Sprite` instead!
+
+                 This class is best used for performance optimization or
+                 internal arcade components.
```

Comment:
I don't think this comment is necessary. We can cover sprites in the programming manual. `BasicSprite` doesn't really work that differently from the normal sprite.

Maybe instead we should mention the limitations. BasicSprite don't support custom hit box and doesn't have rotation. That's basically the difference I think.

Reply:
Maybe take out the warning. Note the major differences. I think BasicSprite for display-only, while Sprite includes collisions? (Not sure that's true, but I think something helping the choice of one vs. the other.)

Reply:
BasicSprite supports basic AABB collision around the entire image. It's geat when you have things like simple tiles or anything that doesn't need any fancy point based collision.

Reply:
> BasicSprite don't support custom hit box 

My understanding is that it does, but with the following restrictions:

1. Only via a `Texture` instance created with a `hit_box_points specified` value since it copies the hitbox from the passed `Texture`
https://github.com/pythonarcade/arcade/blob/ee1bb2b4c49e2287ee4c1e4724a81d6c73390eb1/arcade/sprite/base.py#L55-L57

2. The hitbox points can't be changed afterward since the `texture` setter only swaps the `Texture` and not the actual local hitbox values
https://github.com/pythonarcade/arcade/blob/ee1bb2b4c49e2287ee4c1e4724a81d6c73390eb1/arcade/sprite/base.py#L385-L401

Reply:
oh right. I think @Cleptomania very recently changed this with the new hit box types.

Reply:
`BasicSprite` doesn't have rotation at least.